### PR TITLE
feat(domain): deleted not part of the data model

### LINKED ISSUE #222: Move deleted out of the data model

### PR DESCRIPTION
**Pull request by basmasking**
Fixes #222

Changes proposed in this pull request:
- made distinction between remove and erase data
- fixed inconsistency with setting of date value
- fixed inconsistency with querying relations

@petermasking and @JohnatMasking I like both of your reviews for this pull request.

### CODE REVIEW

**Review by petermasking — CHANGES_REQUESTED** [review]

File: src/domain/reaction/data/mapRecord.ts [hunk 1]
```diff
@@ -12,7 +12,6 @@ export default function mapRecord(record: RecordData): ReactionData
         record.comicId as string,
         record.commentId as string,
         record.ratingCount as number,
-        new Date(record.createdAt as string),
-        record.deleted as boolean,
+        new Date(record.createdAt as string)
```
[the remark this answers is not in the document]
Comment (basmasking) [reply src/domain/reaction/data/mapRecord.ts:15]: @petermasking as discussed, we will use a string for all dates. So this Date change will be reverted and other parts of the data model will be updated with the ISO Date string values.

File: src/domain/reaction/data/remove.ts [hunk 2]
```diff
@@ -2,8 +2,11 @@
 import database from '^/integrations/database/module';
 
 import { RECORD_TYPE } from '../definitions/constants';
+import type ReactionData from './ReactionData';
 
-export default async function remove(id: string): Promise<void>
+export default async function remove(post: ReactionData): Promise<void>
 {
-    return database.deleteRecord(RECORD_TYPE, id);
+    const updatedData = { ...post, deleted: true };
```
Comment (petermasking) [inline src/domain/reaction/data/remove.ts:9]:
```suggestion
    const updatedData: RecordData = { ...post, deleted: true };
```

😉

**Review by petermasking — CHANGES_REQUESTED** [review]

**Review by petermasking — APPROVED** [review]

**Review by JohnatMasking — COMMENTED** [review]
All clear.

**Review by JohnatMasking — APPROVED** [review]
All agreed. No comments

**Pull request merged**